### ISSUE TITLE
Adding Coiled to list of options for setting up Dask

### ISSUE THREAD

**Issue by koverholt** [dask/community#166]
This [Dask documentation page](https://docs.dask.org/en/latest/setup.html) lists various methods for setting up Dask on a single machine or distributed clusters. [Coiled](https://coiled.io/) offers functionality to provision hosted Dask clusters similar to other methods in that list.

Many Dask users ask about [all the different ways Dask users can deploy a Dask cluster](https://twitter.com/rabernat/status/1404834427931353089). Other Coiled users mention that they tried a number of different methods listed on the setup page before discovering Coiled as an easier alternative that handles the backend infrastructure.

I'd like to add Coiled to the `Distributed computing` section on that page. However, I wanted to open this issue for discussion first to consider:

- How to handle linking to commercial solutions on that Dask setup page (if at all)?
- If/how we want to differentiate solutions that are OSS vs. commercial?

**Comment by brl0** [comment]
IMO, that seems reasonable.  I think there is at least some precedence considering that there are a couple of pages that mention Prefect (for example, [ETL Pipelines with Prefect](https://examples.dask.org/applications/prefect-etl.html) and [Prefect: Production Workflows](https://stories.dask.org/en/latest/prefect-workflows.html)).  

Another consideration for inclusion in the list might be [qhub](https://github.com/quansight/qhub).

I am not sure about the specific page or list, which I personally find a bit confusing or inconsistent in what options are listed (for example, dask-gateway is not listed, and the linked kubernetes [page](https://docs.dask.org/en/latest/setup/kubernetes.html) has instructions for deploying the daskhub helm chart, but still not for dask-gateway itself).  So I wonder if somewhere else, or a new page altogether, would be a more appropriate location to add such information.

I could also see considering having an additional commercial services and support page as well, but can also appreciate if this is possibly controversial or divisive or perhaps disincentivizing for some contributors.

**Comment by TomAugspurger** [comment]
Thanks for raising this issue Kris (also 👋, it's been a while).

It's great that we have multiple organizations offering hosted Dask deployments, and I don't have any problem giving them some room in the Dask docs. IMO, a new page under Setup with "hosted Dask" or something similar would fit in well. I don't think we need to differentiate between free / OSS and paid / closed-source offerings.

It's also worth thinking through what we would require from the projects that are included there:

- Regular attendance at the maintainers meeting from an employee (too extreme?)
- A moderately up-to-date version of Dask is offered, if the organization is offering some form of environments

> for example, dask-gateway is not listed

That's probably just an oversight. It'd probably fit well as a top-level page, with a focus on it being centrally managed. Other pages (like Kubernetes and Hadoop) could link to it.

**Comment by jacobtomlinson** [comment]
Agreed that not including Gateway is just an accident and we should definitely update it.

I think listing commercial Dask offerings is a good thing to do. I'm torn about @TomAugspurger's point on insisting orgs regularly attend the maintainers meeting. On one had it feels like advertising and it would be nice to get some guaranteed maintenance time in exchange. On the other the easier it is to use Dask the better it is for the Dask community, so even if folks are not contributing upstream there is still value in telling people about it.

**Comment by TomAugspurger** [comment]
Agreed with how you stated tradeoff Jacob.

We also have the section at https://dask.org/#supported-by, which is where we recognize companies supporting Dask development, so that's already covered to an extent.

**Comment by jsignell** [comment]
I am in favor of there not being a maintenance requirement to be listed on the page that Kris linked. _But_ the descriptions should be very short like what is on the page now.

**Comment by martindurant** [comment]
I do think that there is a line to be drawn between projects that are directly supported by the dask community - more or less the same as being a repo in the dask org - and those that are provided externally. For example, it defines whether they have docs in the same namespace and look'n'feel, and comes with an unspecified promise from the maintenance team: if one of these projects were to loose it's maintainers, but a critical bug arose, one of us would fix it.

For the docs list being talked about here, I would include the ones we directly support separately from others. Even wording like "dask as a service is available from some commercial providers" and link to a separate page.

The outlier is actually qhub - free and open source, but with optional paid support - and, as mentioned, prefect (which is something else again).

I would list sections:
- "companies that support dask development" (requires maintainer salary and attendance)
- "companies that provide paid support"  (on request, if company is in good standing)
- "private companies that build services around dask"  (on request, if dask is mentioned prominently on their page)

None of the above would be on the "how to deploy" page.

Just my suggestion...

**Comment by martindurant** [comment]
^ those sections can, of course, have overlap; and I recognise that significant sections of the current maintenance crew is employed by the various support or deploy companies.

**Comment by koverholt** [comment]
Thanks all for the feedback and input here! Also 👋 @TomAugspurger and many other familiar faces, it has definitely been a while. 😄 

It sounds like there are a few possible approaches here:

1. Add an item to the list in the `Distributed computing` section of the [Dask Setup](https://docs.dask.org/en/latest/setup.html) that links to a `Hosted Dask clusters` page, which then lists various solutions such as Coiled.
2. Add a page with distinct sections for "Companies that offer X", where X is paid support, hosted Dask clusters, or other offerings. Coiled would fit under hosted Dask clusters on such a page.
3. Add an item to the list in the `Distributed computing` section of the [Dask Setup](https://docs.dask.org/en/latest/setup.html) that lists Coiled as an option and has a very short description in line with the other ~10-word descriptions on that page.
4. Something else!

For Approach 1, my feeling is that such a new page might end up repeating a lot of what is already on external sites like Coiled and take too much attention away from the existing Dask deployment solutions. I agree with @jsignell that these "descriptions should be very short like what is on the page now", and I feel like a separate page for hosted Dask clusters would violate that principle and turn mostly into a marketing page.

For Approach 2, as most folks here pointed out, there are existing sections in the Dask docs for sponsors and paid support. It might be helpful to list all of these in one page at some point, but that seems outside of the scope of this particular request. Most of the feedback that I hear from Coiled users is that they looked specifically at the [Dask setup](https://docs.dask.org/en/latest/setup.html) page and tried one or more options there before even realizing that there were other managed/hosted offerings for setting up Dask clusters. Given where users are going today to find solutions here, it seems that it would be better for such hosted Dask offerings to be listed directly in the context of and next to alternative deployment tools rather than a separate companies or commercial offerings page that they would not have visited.

For Approach 3, I think this solves the immediate need of listing Coiled as one of many options/alternatives to deploy Dask and builds on the list of existing solutions there without introducing anything wildly different/new here. If folks agree, I can open a simple PR that adds a link to Coiled and a very short description to the `Distributed computing` section on the Dask setup page, and reference this issue from that PR.

**Comment by jsignell** [comment]
A PR always makes the conversation more concrete, but in the meantime here's one more idea:

4. Add a single bullet to "Distributed computing" that is called "Managed" or something similar and do sub-bullets under that for each Saturn, Coiled, (maybe QHub)...

**Comment by koverholt** [comment]
Opened https://github.com/dask/dask/pull/7814 based on the discussion here. Thanks!

As noted in the PR, I initially added a `Managed / hosted Dask clusters` section under the `Distributed computing` section, but the formatting looked odd compared to the rest of the list.

Also, someone other than me should probably write the descriptions for Saturn and/or QHub, so I'll defer that for now and leave it out of scope of my PR. 🙂

**Comment by jacobtomlinson** [comment]
Given that https://github.com/dask/dask/pull/7814 has been merged I'm going to close this out.
